synapse/api/room_versions.py

```diff
@@ -234,12 +248,12 @@ class RoomVersionCapability:
     for cap in (
         RoomVersionCapability(
             "knock",
-            RoomVersions.V7,
+            RoomVersions.V8,
             lambda room_version: room_version.msc2403_knocking,
         ),
         RoomVersionCapability(
             "restricted",
-            None,
+            RoomVersions.V8,
             lambda room_version: room_version.msc3083_join_rules,
         ),
```

Comment:
this probably wants to be done at in a subsequent release otherwise clients will advertise this new feature to users with high likelihood that an upgrade to v8 will block existing members from joining the upgraded room until a decent chunk of Synapses support v8

Reply:
Hmm...yeah I guess that makes sense. I guess there isn't a better way to do this, it is similar to updating the default version but we can probably be a bit more aggressive.